src/main/java/org/quiltmc/asmr/processor/capture/AsmrReferenceNodeCapture.java

```diff
@@ -48,6 +48,27 @@ public String className() {
         return path;
     }
 
+
+    @Override
+    public int startVirtualIndex() {
+        return Integer.MIN_VALUE;
+    }
+
+    @Override
+    public int endVirtualIndex() {
+        return Integer.MAX_VALUE;
+    }
+
+    @Override
+    public int startIndexInclusive() {
+        return Integer.MIN_VALUE;
```

Comment:
We have to be a bit careful here that we never end up multiplying this value by 2. We might have to think a little harder about what exactly these values should be.

Reply:
Yeah, I considered that too. Could set it to `Integer.MIN_VALUE / 2`. Or even make `startIndexInclusive()` a default method that halves `startVirtualIndex`.

Reply:
It might even be better for them to be even smaller. The virtual start index can be as high as `-1` however the actual index should then maybe be calculated using `>> 1`. For the max, we could go 1 higher than the max virtual index possible if it were a slice (so `2 * children().size() + 2`).

Reply:
Yeah that sounds like a good idea

Reply:
What sounds like a good idea? the halving, the extracting to a default method, using `>> 1` instead of `/ 2`, or reducing the value of the min and max

Reply:
The problem with `2 * children().size() + 2` is if you are making a node capture of a list, node captures aren't shifted. So I think `Integer.MAX_VALUE - 2`, `Integer.MAX_VALUE / 2 - 1`, `-1` and `-2` are good enough for now.